include/dgl/runtime/ndarray.h

```diff
@@ -130,7 +130,7 @@ class NDArray {
   /*! \return the data pointer with type. */
   template <typename T>
   inline T* Ptr() const {
-    if (!defined())
+    if (!defined() || operator->()->shape[0] == 0)
```

Comment:
I don't think that's a good idea. `Ptr` should just return the internal data pointer.

Reply:
```operator->()->shape[0] == 0``` is the condition that an NDArray is NullArray.
In some kernel we need this behaviour (e.g. use `edge_map != nullptr` to decide whether edge mapping is required or not).

Reply:
Removed all dependencies.